src/java/org/apache/cassandra/net/InboundConnectionSettings.java

```diff
@@ -157,7 +157,8 @@ public InboundConnectionSettings withLegacyDefaults()
         ServerEncryptionOptions encryption = this.encryption;
         if (encryption == null)
             encryption = DatabaseDescriptor.getInternodeMessagingEncyptionOptions();
-        encryption = encryption.withOptional(false);
+        encryption = encryption.withOptional(false).withInternodeEncryption(ServerEncryptionOptions.InternodeEncryption.all);
```

Comment:
in 3.0 I see the default is `none` and not `all`

Reply:
The `withLegacyDefaults` template is used if the legacy ssl storage port is enabled, and connections on that port should always be encrypted so I defaulted to `all` to make sure the SSL handler (and not the optional SSL handler) was installed.